### PR TITLE
feat/wire-up new backend endpoints to support multiple flows

### PR DESCRIPTION
**Pull request by lordrip**
### Context
With the new endpoints introduced by https://github.com/KaotoIO/kaoto-backend/pull/638, we could start wiring up the source code editor and the related components to use the multiple flows approach.

### Changes
This pull request replaces the previous `v1/integrations` endpoint with the new `v2/integrations` endpoint that supports multiple flows.

All the changes were validated against the existing e2e tests to ensure that the previous functionality from the single flow is still in place.

### Known issues
At this stage, we cannot create new flows through the canvas, although writing multiple flows from the source code is possible.

#### SIngle flow
image

#### Multiple flows
image

Relates to: https://github.com/KaotoIO/kaoto-ui/issues/801

### CODE REVIEW

File: src/components/SettingsModal.tsx [hunk 1]
```diff
@@ -51,39 +56,60 @@ export const SettingsModal = ({ handleCloseModal, isModalOpen }: ISettingsModal)
 
   const { addAlert } = useAlert() || {};
 
+  const onChangeDsl = useCallback((value: string) => {
```
Comment (igarashitm) [inline src/components/SettingsModal.tsx:59]: Maybe in a separate one, but might be better to create a service and move this into it, like `FlowsService`

File: src/components/SourceCodeEditor.tsx [hunk 2]
```diff
@@ -24,61 +25,65 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
...
-
-  const fetchTheSourceCode = (integration: IIntegration, settings: ISettings) => {
-    const tmpIntegration = {
-      ...integration,
-      metadata: { ...integrationJson.metadata, ...settings },
-      dsl: settings.dsl.name,
+  }, [flows, properties]);
+
+  const fetchTheSourceCode = (currentFlowsWrapper: IFlowsWrapper, settings: ISettings) => {
+    const updatedFlowWrapper = {
```
Comment (igarashitm) [inline src/components/SourceCodeEditor.tsx:55]: And this one is also a candidate in `FlowsService`, then can we put all YAML>JSON and JSON>YAML work in a single service?

File: src/components/SourceCodeEditor.tsx [hunk 3]
```diff
@@ -24,61 +25,65 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
...
-  const previousJson = usePrevious(integrationJson);
+  const { flows, properties, setFlows } = useFlowsStore(({ flows, properties, setFlows }) => ({ flows, properties, setFlows }), shallow);
+  const previousFlows = usePrevious(flows);
+
   const schemaUri = settings.dsl.validationSchema
     ? process.env.KAOTO_API + settings.dsl.validationSchema
     : '';
 
   useEffect(() => {
-    let tmpInt: IIntegration = integrationJson;
```
Comment (igarashitm) [inline src/components/SourceCodeEditor.tsx:35]: And the content in this `useEffect` would be also a candidate to move out to a `FlowsService`

**Review by igarashitm — DISMISSED** [review]
Approving - some routines in the components jumped out for me, looks like a needs of a service, but not directly related to this change

File: src/components/SourceCodeEditor.tsx (hunk 2 above)
Comment (lordrip) [reply src/components/SourceCodeEditor.tsx:55]: Yup, that makes sense :ok_hand:

**Pull request merged**